### PR TITLE
R.scan now dispatches to scan method.

### PR DESCRIPTION
**Pull request by jethrolarson**
Fixes half of #1179

### CODE REVIEW

**Comment by jethrolarson** [comment]
Ping

File: src/scan.js [hunk 1]
```diff
@@ -21,7 +23,12 @@ var _curry3 = require('./internal/_curry3');
  *      var factorials = R.scan(R.multiply, 1, numbers); //=> [1, 1, 2, 6, 24]
  */
 module.exports = _curry3(function scan(fn, acc, list) {
-  var idx = 0, len = list.length, result = [acc];
+  if (!_isArray(list) && typeof list.scan === 'function') {
```
Comment (davidchambers) [inline src/scan.js]: Is `!_isArray(list)` necessary?

File: src/scan.js [hunk 2]
```diff
@@ -1,10 +1,12 @@
 var _curry3 = require('./internal/_curry3');
+var _isArray = require('./_isArray');
 
 
 /**
  * Scan is similar to reduce, but returns a list of successively reduced values
  * from the left
  *
+ * Dispatches to the `scan` method of the third argument, if present.
  * @func
```
Comment (davidchambers) [inline src/scan.js]:
```diff
  * Dispatches to the `scan` method of the third argument, if present.
+ *
  * @func
```

File: src/scan.js (hunk 1 above)
Comment (jethrolarson) [inline src/scan.js]: I suppose not, if people are modifying the array prototype they might be doing so knowingly. I know we intercept it this way for reduce

Comment (davidchambers) [inline src/scan.js]: I'd prefer to remove the guard, in that case. It strikes me as an unnecessary exception to this rule:

> Dispatches to the `scan` method of the third argument, if present.

**Comment by CrossEye** [comment]
While this might be worth doing, we would need to revisit the implementation these days.  Closing without prejudice.

**Pull request closed without merging**